app/editPlan/getTaskListData.js

```diff
@@ -0,0 +1,107 @@
+const NOT_STARTED = 'NOT STARTED'
+const CANNOT_START_YET = 'CANNOT START YET'
+const INCOMPLETE = 'INCOMPLETE'
+const COMPLETED = 'COMPLETED'
+
+const ADD_DETAILS = 'Add details'
+const REVIEW_DETAILS = 'Review details'
+
+const reviewComments = (comments, commentTypes) => {
+  const statusCount = comments.filter(({ commentType }) => commentTypes.includes(commentType))
+  let statusText = NOT_STARTED
+  let complete = false
+  let hrefText = ADD_DETAILS
+
+  if (statusCount.length === 1) statusText = INCOMPLETE
+  else if (statusCount.length >= 2) {
+    statusText = COMPLETED
+    complete = true
+    hrefText = REVIEW_DETAILS
+  }
+  return { statusText, complete, hrefText }
+}
+
+const getAboutTheIndividual = ({ comments }, stub) => {
+  return {
+    heading: {
+      text: 'Write about the individual',
+    },
+    items: [
+      {
+        text: 'Responsivity',
+        href: `${stub}/review-responsivity`,
+        ...reviewComments(comments, ['YOUR_RESPONSIVITY', 'THEIR_RESPONSIVITY']),
+      },
+    ],
+  }
+}
+const actionsComplete = (
+  complete,
+  { description = '', intervention = '', motivationUUID = '', updated = '', owner = [], ownerOther = '' }
+) =>
+  complete &&
+  (description.length > 0 || intervention.length > 0) &&
+  motivationUUID.length > 0 &&
+  updated.length > 0 &&
+  (owner.length > 0 || ownerOther.length > 0)
+const getAddObjectives = ({ objectives = [] }, stub) => {
+  const addedObjectives = objectives.map(({ actions = [], needs = [], description = '', id }, index) => {
+    const addedObjective = {
+      text: description.length > 0 ? description : `Add objective ${index + 1}`,
+      hrefText: ADD_DETAILS,
+      href: `${stub}/edit-objective/${id}`,
+      statusText: INCOMPLETE,
+      complete: false,
+    }
+    if (description.length === 0 && needs.length === 0) {
+      addedObjective.statusText = NOT_STARTED
+    } else if (description.length > 0 && needs.length > 0 && actions.reduce(actionsComplete, true)) {
```

Comment:
Just checking my understanding - does an objective always need to be linked to at least 1 need?

Reply:
Yes. As of time of writing 😬